src/util/expr_initializer.cpp

```diff
@@ -41,22 +43,13 @@ class zero_initializert:public messaget
 protected:
   const namespacet &ns;
 
-  std::string to_string(const exprt &src)
-  {
-    return expr2c(src, ns);
-  }
-
-  std::string to_string(const typet &src)
-  {
-    return type2c(src, ns);
-  }
-
   exprt zero_initializer_rec(
     const typet &type,
     const source_locationt &source_location);
 };
 
-exprt zero_initializert::zero_initializer_rec(
+template <bool nondet>
+exprt zero_initializert<nondet>::zero_initializer_rec(
   const typet &type,
```

Comment:
It's confusing that zero_initializert does non-det initialisation -- either keep separate, or rename template to expr_initializert.